tests/TestData/Providers/EntitiesFilter.php

```diff
@@ -0,0 +1,80 @@
+<?php
+declare(strict_types=1);
+
+namespace StubTests\TestData\Providers;
+
+use StubTests\Model\BasePHPElement;
+use StubTests\Model\PHPClass;
+use StubTests\Model\PHPFunction;
+use StubTests\Model\PHPInterface;
+use StubTests\Model\StubProblemType;
+
+class EntitiesFilter
+{
+    /**
+     * @param BasePHPElement[] $entities
+     * @param int ...$problemTypes
+     * @return array
+     */
+    public static function getFiltered(array $entities, int ...$problemTypes): array
+    {
+        $resultArray = [];
+        $hasProblem = false;
+        foreach ($entities as $entity) {
+            if (!empty($problemTypes)) {
```

Comment:
This is unnecessary, there will be no iteration over empty list: https://3v4l.org/nCcbY